owncloudApp/src/main/java/com/owncloud/android/presentation/ui/sharing/fragments/ShareFileFragment.kt

```diff
@@ -141,7 +142,7 @@ class ShareFileFragment : Fragment(), ShareUserListAdapter.ShareUserAdapterListe
             val usedNumbers = ArrayList<Int>()
             var isDefaultNameSet = false
             var number: String
-            for (share in publicLinks as ArrayList<OCShareEntity>) {
+            for (share in publicLinks as ArrayList<OCShare>) {
```

Comment:
why are you using this casting??

In addition, in case you need to cast, do it to List<OCShare> instead

Reply:
Done